pysparkling/utils.py

```diff
@@ -346,22 +347,30 @@ def strhash(string):
     return x
 
 
-def tuplehash(tuple):
+def portable_hash(x):
     """
-    Old python hash function for tuples.
-
-    :param tuple: tuple to hash
-    :return: hash
+    This function returns consistent hash code for builtin types, especially
+    for None and tuple with None.
+    The algorithm is similar to that one used by CPython 2.7
+    >>> portable_hash(None)
+    0
+    >>> portable_hash((None, 1)) & 0xffffffff
+    219750521
     """
-    if isinstance(tuple, int):
-        return tuple
-    mult = 1000003
-    x = 0x345678
-    for index, item in enumerate(tuple):
-        x = ((x ^ hash(item)) * mult) & (1<<32)
-        mult += (82520 + (len(tuple)-index)*2)
-    return x + 97531
-
-
-def portable_hash(item):
-    return strhash(item) & 0xffffffff
+
+#    if sys.version_info >= (3, 2, 3) and 'PYTHONHASHSEED' not in os.environ:
+#        raise Exception("Randomness of hash of string should be disabled via PYTHONHASHSEED")
+
+    if x is None:
+        return 0
+    if isinstance(x, tuple):
+        h = 0x345678
+        for i in x:
+            h ^= portable_hash(i)
+            h *= 1000003
+            h &= sys.maxsize
+        h ^= len(x)
+        if h == -1:
+            h = -2
+        return int(h)
+    return strhash(str(x))
```

Comment:
Can't we return hash(x) if x is not a string? I believe the need of strhash is only here when we deal with str

Doesn't change much anyway but this way we are a bit more align with pyspark

Reply:
The tests seem to run fine when using `hash()` for non string types. I pushed the changes.